pallets/utxo/src/lib.rs

```diff
@@ -631,8 +668,14 @@ pub mod pallet {
                     Destination::CreatePP(_, _) => {
                         log::info!("TODO validate spending of OP_CREATE");
                     }
-                    Destination::CallPP(_, _) => {
-                        log::info!("TODO validate spending of OP_CALL");
+                    Destination::CallPP(_, _, _) => {
+                        let spend =
+                            u16::from_le_bytes(input.witness[1..].try_into().or_else(|_| {
+                                Err(DispatchError::Other(
+                                    "Failed to convert witness to an opcode",
+                                ))
+                            })?);
+                        ensure!(spend == 0x1337, "OP_SPEND not found");
```

Comment:
AFAIK the first byte denotes the length of the integer, which in this case was `0x02`. The check didn't look quite as nice with that length byte included. But I've realized the implementation of this opcode is incomplete because we don't only have to worry about validation of outgoing transaction but also need to validate incoming transaction. I'll need to redo this and I'll mark this conversation is resolved because this code needs will need another review anyway